test/framework/clusterctl/management_cluster.go

```diff
@@ -1,194 +0,0 @@
-// +build e2e
-
-/*
-Copyright 2020 The Kubernetes Authors.
-
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package clusterctl
-
-import (
-	"bufio"
-	"context"
-	"fmt"
-	"io"
-	"os"
-	"path"
-	"path/filepath"
-
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-
-	appsv1 "k8s.io/api/apps/v1"
-	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime"
-	clusterctlconfig "sigs.k8s.io/cluster-api/cmd/clusterctl/client/config"
-	"sigs.k8s.io/cluster-api/test/framework"
-	"sigs.k8s.io/cluster-api/test/framework/discovery"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-)
-
-// Provides utilities for setting up a management cluster using clusterctl.
-
-// InitManagementClusterInput is the information required to initialize a new
-// management cluster for e2e testing.
-type InitManagementClusterInput struct {
-	// E2EConfig defining the configuration for the E2E test.
-	E2EConfig *E2EConfig
-
-	// ClusterctlConfigPath is the path to a clusterctl config file that points to repositories to be used during "clusterctl init".
-	ClusterctlConfigPath string
-
-	// LogsFolder defines a folder where to store clusterctl logs.
-	LogsFolder string
-
-	// Scheme is used to initialize the scheme for the management cluster client.
-	Scheme *runtime.Scheme
-
-	// NewManagementClusterFn should return a new management cluster.
-	NewManagementClusterFn func(name string, scheme *runtime.Scheme) (cluster framework.ManagementCluster, kubeConfigPath string, err error)
-}
-
-// InitManagementCluster returns a new cluster initialized and the path to the kubeConfig file to be used to access it.
-func InitManagementCluster(ctx context.Context, input *InitManagementClusterInput) (framework.ManagementCluster, string) {
```

Comment:
It looks like this is currently being used by at least cluster-api-provider-vsphere.

Reply:
@detiber this is `test/framework/clusterctl.InitManagementCluster`, while in CAPV they are using `test/framework.InitManagementCluster` (without clusterctl)

Reply:
ah, carry on then :)